### ISSUE TITLE
Address parser bug

### ISSUE THREAD

**Issue by visualwebs** [Paazl/magento2-checkout-widget#51]
Hi 

I can share a possible bug, to improve the module, is related to address parsing from Magento to Paazl.

To reproduce the error let´s to fill shipping address with those values:

street : mystreet 21 2
So we expect those values:

street:mystreet
houseNumber:21
houseNumberExtension:2

See the log output, I replace with some dummy values for privacy:

`AddOrder request: : {"additionalInstruction":null,"consignee":{"email":"myemail@hotmail.com","name":"test test","companyName":null,"phone":null,"address":{"city":"Amsterdam","country":"NL","postalCode":"mytestingpostcode","province":null,"street":"mystreet 21","houseNumber":"2","houseNumberExtension":""}},"customsValue"...`

So the method Paazl\CheckoutWidget\Model\Api\Builder\Order::parseAddress in this case works wrong. It joins street with house number "mystreet 21" as street. And uses house number extension as house number. House number extension appears empty and should be 2.

I created a simple plugin for this purpose that makes the trick. In our case we use single address line and this solution fixes the problem.

public function afterParseAddress(PaazlOrder $subject, $result, $shippingAddress)
    {
       $street = trim(implode(' ', $shippingAddress->getStreet()));
       $street = explode(' ', $street);
       
      if ($street && count($street) == 3)
       {
           if (is_numeric($street[1]) && is_numeric($street[2]))
           {
               return [
                    'street' => $street[0],
                    'houseNumber' => $street[1],
                    'houseNumberExtension' => $street[2],
               ];
           }
        
       }
        
       return $result;
    }

**Comment by Marvin-Magmodules** [comment]
Hi @visualwebs, thank you for opening this issue.
There are some combinations that are hard to parse, so to get the optimal result we recommend using three fields for an address in the checkout and separate house number and house number addition. This will move the validation to the customer who knows best. The plugin you have created will fail if the street contains spaces.

**Comment by visualwebs** [comment]
Thanks for the tips. Currently, we have hundreds of customer addresses in our project and we won´t change the address settings. The plugin I created is working on the website for a long time and resolved the problem for us.
I close this case.
